api/classes/publisher.js

```diff
@@ -1,19 +1,108 @@
-// var AWS = require('aws-sdk');
-// var s3 = new AWS.S3();
+var knox = require('knox');
+var client = knox.createClient({
+  key: process.env.AWS_ACCESS_KEY_ID,
+  secret: process.env.AWS_SECRET_ACCESS_KEY,
+  bucket: process.env.AWS_BUCKET
+});
 
 var log = require('../../logger.js');
+var File = require('../modules/Files/model');
 
 var isPublished = function(project) {
   return project.get('publish_url') !== null;
 };
 
-exports.publish = function(project) {
-  var url = 'FAKE://URL';
+var publishFiles = function(project, params) {
+  File.query({
+    where: {
+      project_id: project.id
+    }
+  }).fetchAll()
+  .then(function(records){
+    records.mapThen(function(record) {
+      uploadFile(record, params);
+    });
+  }).catch(function(e) {
+    log.error(e);
+    return false;
+  });
+  return true;
```

Comment:
This will always return true, since the `return false` on line 27 is inside a callback function.

Maybe try something like:

```js
return File.query(...)
  .then(fn() {
    // ... 
    return Promise.resolve(true); 
  }).catch(fn(e) {
    return Promise.reject(false);
  });
```